---
output: github_document
---

## Workshops

Join the _Bioconductor_  [#bioc2020](https://community-bioc.slack.com/archives/CLAEUFVAA)
and [#bioc2020-workshops](https://community-bioc.slack.com/archives/C015QR5FCGN)
channels for up-to-date information.

```{r,eval=FALSE,echo=FALSE}
# run this to create the workshops.md file
rmarkdown::render("workshops.Rmd",
    rmarkdown::md_document(
        variant="gfm", preserve_yaml = FALSE
    )
)
writeLines(c(
"---
layout: default
---

{% include header.md %}
",

    readLines("workshops.md")
    ),
    con = file("workshops.md")
)
```

```{r,results="hide",echo=FALSE,message=FALSE,warning=FALSE}
.github <- "https://raw.githubusercontent.com"
.dock_hub <- "https://hub.docker.com/repository/docker/"

make_repo_table <- function(repos) {
    descs <- file.path(.github, repos, "master/DESCRIPTION")
    repo_url <- file.path("https://github.com", repos)
    dcfs <- lapply(descs, function(desc_url) {
        tryCatch({
            read.dcf(url(desc_url))
            }, error = function(e) conditionMessage(e)
        )
    })
    valid <- vapply(dcfs, function(x) is.matrix(x), logical(1L))
    noDESC <- repos[!valid]
    msg <- paste0("Missing `DESCRIPTION`: ", paste0(noDESC, collapse = ", "))
    noDESC <- if (!length(noDESC)) "" else msg
    dcfs <- dcfs[valid]
    annote <- dplyr::bind_rows(
        lapply(lapply(dcfs, as.data.frame), tibble)
    )
    updf <- annote[, c("Title", "URL", "DockerImage")]
    docks <- lapply(strsplit(updf[["DockerImage"]], ":"), `[`, 1L)
    workshop <- paste0("[", updf[["Title"]], "]", "(", updf[["URL"]], ")")
    github <- paste0("[![GH](images/GitHub-Mark-32px.png)]", "(", repo_url[valid], ")")
    docker <- paste0("[![Docker](images/docker_icon.png)]", "(", .dock_hub, docks, ")")
    list(
        tibble(workshop, github, docker),
        noDESC
    )
}
```

### 100 Level Workshops

```{r,echo=FALSE,message=FALSE,warning=FALSE}
library(knitr)
library(kableExtra)
library(tibble)
l100 <- c(
    "jmacdon/Bioc2020Anno",
    "stemangiola/bioc_2020_tidytranscriptomics",
    "waldronlab/PublicDataResources",
    "databio/cocoa_workshop_bioc2020",
    "LieberInstitute/recountWorkshop2020",
    "waldronlab/AnVILWorkshop",
    "cmirzayi/EpiForBioWorkshop2020"
)
rtab <- make_repo_table(l100)
knitr::kable(rtab[[1]], format = "markdown",
    col.names = c("Workshop", "Source", "Docker"))
```

```{r, echo=FALSE,results="asis"}
if (length(rtab[[2]]))
    cat(
        paste0("<sup>", rtab[[2]],  "</sup>", sep = "\n")
    )
```

### 200 Level Workshops

```{r,echo=FALSE,message=FALSE,warning=FALSE}
l200 <- c(
    "haibol2016/ATACseqQCWorkshop",
    "iSEE/iSEEWorkshop2020",
    "waldronlab/enrichOmics",
    "waldronlab/CNVWorkshop",
    "waldronlab/MultiAssayWorkshop",
    "LihuaJulieZhu/CRISPRseekGUIDEseqBioc2020Workshop/",
    "hukai916/IntegratedChIPseqWorkshop",
    "aedin/PCAworkshop",
    "bhklab/bioc2020workshop",
#    "broadinstitute/BioC2020Workshop_AMARETTO-Hub",
    "mikelove/alevin2bioc"
)
rtab <- make_repo_table(l200)
knitr::kable(rtab[[1]], align = "l", escape = FALSE,
    format = "markdown", col.names = c("Workshop", "Source", "Docker"))
```

```{r, echo=FALSE,results="asis"}
if (length(rtab[[2]]))
    cat(
        paste0("<sup>", rtab[[2]],  "</sup>", sep = "\n")
    )
```

### 500 Level Workshops

```{r,echo=FALSE,message=FALSE,warning=FALSE}
l500 <- c(
    "Kayla-Morrell/CreateAPackage",
    "Liubuntu/Bioc2020RCWL",
    "PeteHaitch/BioC2020_DelayedArray_workshop",
    "mdozmorov/HiCcompareWorkshop",
    "kstreet13/bioc2020trajectories",
    "montilab/hypeR-workshop"
)
rtab <- make_repo_table(l500)
knitr::kable(rtab[[1]], align = "l", escape = FALSE,
    format = "markdown", col.names = c("Workshop", "Source", "Docker"))
```

```{r, echo=FALSE,results="asis"}
if (length(rtab[[2]]))
    cat(
        paste0("<sup>", rtab[[2]],  "</sup>", sep = "\n")
    )
```

Workshops with preinstalled dependencies can be run locally using Docker. 

- `docker pull <userid/workshop:latest>` will pull the latest image for a given `<workshop>`. 
- `docker run -e PASSWORD=<yourpassword> -p 8787:8787 -d --rm <userid/workshop>` will publish a container's port 8787 to the host (`-p`), run in the detached mode (`-d`), and cleanly remove the container when it is stopped (`--rm`). 
- Open [http://localhost:8787](http://localhost:8787) and login with username `rstudio` and password `<yourpassword>`. 
- Run `browseVignettes(package = "<workshop>")`. Click on one of the links, "HTML", "source", "R code".
    - In case of "`The requested page was not found`" error, add `help/` to the URL right after the hostname. This is a [known bug](https://github.com/rocker-org/rocker-versioned/issues/178).

Advanced: Use `-v $(pwd):/home/rstudio` argument to map your local directory to the container. Use `-e DISABLE_AUTH=true`, if you want passwordless login to RStudio.  On Windows, you may need to provide your localhost IP address like `http://191.163.92.108:8787/` - find it using `docker-machine ip default` in Docker's terminal.

Example:

```
docker pull waldronlab/publicdataresources:latest
docker run -e PASSWORD=bioconductor -p 8787:8787 -d --rm waldronlab/publicdataresources
# Open http://localhost:8787 and login using rstudio/bioconductor credentials
# Run browseVignettes(package = "PublicDataResources")
# Open http://localhost:8787/help/library/PublicDataResources/doc/PublicDataResources.html
docker ps -a # List all running containers
docker stop <CONTAINER ID> # or, <NAMES> - Stop a container
```

Workshop packages were created using the
[BuildABiocWorkshop2020](https://github.com/seandavi/BuildABiocWorkshop2020)
template.

*NOTE*. All workshops should include a valid `DESCRIPTION` file with the
following fields:

* `URL:` - the GitHub pages URL (`https://username.github.io/repository`)
* `DockerImage:` - the Docker repository slug in lowercase with (optional) tag
(`username/repository:tag`)
