---
layout: default
---

{% include header.md %}

## Workshops

Join the _Bioconductor_  [#bioc2020](https://community-bioc.slack.com/archives/CLAEUFVAA) and [#bioc2019-workshops](https://community-bioc.slack.com/archives/CJDMYKG2U) channels for up-to-date information

| 100 Level Workshops |
|---------------------|
| [Introduction to Bioconductor Annotation Resources](https://jmacdon.github.io/Bioc2020Anno/) | ![](https://github.com/jmacdon/Bioc2020Anno/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/jmacdon/Bioc2020Anno) | [Docker](https://hub.docker.com/repository/docker/jmacdon/Bioc2020Anno) |
| [A Tidy Transcriptomics introduction to RNA-Seq analyses](https://stemangiola.github.io/bioc_2020_tidytranscriptomics/) | ![](https://github.com/stemangiola/bioc_2020_tidytranscriptomics/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/stemangiola/bioc_2020_tidytranscriptomics) | [Docker](https://hub.docker.com/repository/docker/stemangiola/bioc_2020_tidytranscriptomics) |
| [Multi-omic Integration and Analysis of cBioPortal and TCGA data with MultiAssayExperiment](https://waldronlab.github.io/MultiAssayWorkshop/) | ![](https://github.com/waldronlab/MultiAssayWorkshop/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/waldronlab/MultiAssayWorkshop) | [Docker](https://hub.docker.com/repository/docker/waldronlab/MultiAssayWorkshop) |
| [Public data resources and Bioconductor](https://waldronlab.github.io/PublicDataResources/) | ![](https://github.com/waldronlab/PublicDataResources/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/waldronlab/PublicDataResources) | [Docker](https://hub.docker.com/repository/docker/waldronlab/PublicDataResources) |
| [Annotation of inter-sample epigenetic variation with COCOA](https://databio.github.io/cocoa_workshop_bioc2020/) | ![](https://github.com/databio/cocoa_workshop_bioc2020/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/databio/cocoa_workshop_bioc2020) | [Docker](https://hub.docker.com/repository/docker/databio/cocoa_workshop_bioc2020) |
| [Human RNA-seq data from recount2 and related packages](https://LieberInstitute.github.io/recountWorkshop2020/) | ![](https://github.com/LieberInstitute/recountWorkshop2020/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/LieberInstitute/recountWorkshop2020/) | [Docker](https://hub.docker.com/repository/docker/LieberInstitute/recountWorkshop2020) |

| 200 Level Workshops | 
|---------------------|
| [Best practices for ATAC-seq QC and data analysis](https://haibol2016.github.io/ATACseqQCWorkshop/) | ![](https://github.com/haibol2016/ATACseqQCWorkshop/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/haibol2016/ATACseqQCWorkshop) | [Docker](https://hub.docker.com/repository/docker/haibol2016/ATACseqQCWorkshop) |
| [iSEE: Interactive visualization of SummarizedExperiment objects](https://iSEE.github.io/iSEEWorkshop2020/) | ![](https://github.com/iSEE/iSEEWorkshop2020/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/iSEE/iSEEWorkshop2020) | [Docker](https://hub.docker.com/repository/docker/iSEE/iSEEWorkshop2020) |
| [Functional enrichment analysis of high-throughput omics data](https://waldronlab.github.io/waldronlab/) | ![](https://github.com/waldronlab/enrichOmics/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/waldronlab/enrichOmics) | [Docker](https://hub.docker.com/repository/docker/waldronlab/enrichOmics) |
| [Copy number variation analysis with Bioconductor](https://waldronlab.github.io/CNVWorkshop/) | ![](https://github.com/waldronlab/CNVWorkshop/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/waldronlab/CNVWorkshop) | [Docker](https://hub.docker.com/repository/docker/waldronlab/CNVWorkshop) |
| [Comprehensive RNAseq Analysis with oneStopRNAseq](https://radio1988.github.io/oneStopRNAseqWorkshop/) | ![](https://github.com/radio1988/oneStopRNAseqWorkshop/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/radio1988/oneStopRNAseqWorkshop) | [Docker](https://hub.docker.com/repository/docker/radio1988/oneStopRNAseqWorkshop) |
| [Integrated ChIP-seq Data Analysis Workshop](https://hukai916.github.io/IntegratedChIPseqWorkshop/) | ![](https://github.com/hukai916/IntegratedChIPseqWorkshop/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/hukai916/IntegratedChIPseqWorkshop) | [Docker](https://hub.docker.com/repository/docker/hukai916/IntegratedChIPseqWorkshop) |
| [Hitchhikers Guide to PCA; Demystifying dimensionality reduction in R/Bioconductor](https://aedin.github.io/Frontiers_Supplement/) | ![](https://github.com/aedin/Frontiers_Supplement/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/aedin/Frontiers_Supplement) | [Docker](https://hub.docker.com/repository/docker/aedin/Frontiers_Supplement) |
| [Epidemiology for Bioinformaticians](https://cmirzayi.github.io/epiforbioworkshop/) | ![](https://github.com/cmirzayi/epiforbioworkshop/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/cmirzayi/epiforbioworkshop) | [Docker](https://hub.docker.com/repository/docker/cmirzayi/epiforbioworkshop) |
| [Biomarker discovery from high throughput screening datasets](https://bhklab.github.io/bioc2020workshop/) | ![](https://github.com/bhklab/bioc2020workshop/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/bhklab/bioc2020workshop) | [Docker](https://hub.docker.com/repository/docker/bhklab/bioc2020workshop) |


| 500 Level Workshops | 
|---------------------|
| [Create A Package](https://Kayla-Morrell.github.io/CreateAPackage/) | ![](https://github.com/Kayla-Morrell/CreateAPackage/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/Kayla-Morrell/CreateAPackage) | [Docker](https://hub.docker.com/repository/docker/Kayla-Morrell/CreateAPackage) |
| [Connecting Bioconductor to other bioinformatics tools using Rcwl](https://Liubuntu.github.io/Bioc2020RCWL/) | ![](https://github.com/Liubuntu/Bioc2020RCWL/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/Liubuntu/Bioc2020RCWL) | [Docker](https://hub.docker.com/repository/docker/Liubuntu/Bioc2020RCWL) |
| [Effectively using the DelayedArray framework as a user to support the analysis of large datasets](https://PeteHaitch.github.io/BioC2020_DelayedArray_workshop/) | ![](https://github.com/PeteHaitch/BioC2020_DelayedArray_workshop/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/PeteHaitch/BioC2020_DelayedArray_workshop) | [Docker](https://hub.docker.com/repository/docker/PeteHaitch/BioC2020_DelayedArray_workshop) |
| [Introduction to the Terra/AnVIL Cloud-based Genomics Platform](https://waldronlab.github.io/AnVILWorkshop/) | ![](https://github.com/waldronlab/AnVILWorkshop/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/waldronlab/AnVILWorkshop) | [Docker](https://hub.docker.com/repository/docker/waldronlab/AnVILWorkshop) |
| [multiHiCcompare: Differential Hi-C Data Analysis Workshop](https://mdozmorov.github.io/HiCcompareWorkshop/) | ![](https://github.com/mdozmorov/HiCcompareWorkshop/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/mdozmorov/HiCcompareWorkshop) | [Docker](https://hub.docker.com/repository/docker/mdozmorov/HiCcompareWorkshop) |
| [Trajectory inference across conditions: differential expression and differential progression](https://kstreet13.github.io/bioc2020trajectories/) | ![](https://github.com/kstreet13/bioc2020trajectories/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/kstreet13/bioc2020trajectories) | [Docker](https://hub.docker.com/repository/docker/kstreet13/bioc2020trajectories) |
| [Efficient, Scalable, and Reproducible Enrichment Workflows](https://montilab.github.io/hypeR-workshop/) | ![](https://github.com/montilab/hypeR-workshop/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/montilab/hypeR-workshop) | [Docker](https://hub.docker.com/repository/docker/montilab/hypeR-workshop) |

Workshop packages were created using the [BuildABiocWorkshop2020](https://github.com/seandavi/BuildABiocWorkshop2020) template



<!--
| 100 Level Workshops | 
|---------------------|
| [](https://XXX.github.io/YYY/) | ![](https://github.com/XXX/YYY/workflows/.github/workflows/basic_checks.yaml/badge.svg) | [GitHub](https://github.com/XXX/YYY) | [Docker](https://hub.docker.com/repository/docker/XXX/YYY) |
-->



<!--
- [Main Conference](#main-conference)
- [Developer Day](#developer-day)

### Main Conference

Workshops will be announced in April based on submissions to the ([call for workshops, talks, posters, and travel awards][call]). See the [BioC2018 workshops][bioc2018workshops] as a guide. Bioconductor will offer the "100-level" courses every year:

- _100: R and [Bioconductor][] for everyone: an introduction_
- _101: Introduction to [Bioconductor][] annotation resources_. 
- _102: Solving common bioinformatic challenges using [GenomicRanges][]_.
- _103: Public Data Resources and [Bioconductor][]_.


[call]: ./travel-accommodations
[bioc2018workshops]: https://bioconductor.github.io/BiocWorkshops/
[Bioconductor]: https://bioconductor.org/
[GenomicRanges]: https://bioconductor.org/packages/GenomicRanges
-->