---
layout: default
---

{% include header.md %}

## Workshops

Join the *Bioconductor*
[\#bioc2020](https://community-bioc.slack.com/archives/CLAEUFVAA) and
[\#bioc2020-workshops](https://community-bioc.slack.com/archives/C015QR5FCGN)
channels for up-to-date information.

### 100 Level Workshops

| Workshop                                                                                                                 | Source                                                                                             | Docker                                                                                                                  |
| :----------------------------------------------------------------------------------------------------------------------- | :------------------------------------------------------------------------------------------------- | :---------------------------------------------------------------------------------------------------------------------- |
| [Introduction to Bioconductor annotation resources](https://jmacdon.github.io/Bioc2020Anno/)                             | [![GH](images/GitHub-Mark-32px.png)](https://github.com/jmacdon/Bioc2020Anno)                      | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/jmacdon/bioc2020anno)                      |
| [A Tidy Transcriptomics introduction to RNA-Seq analyses](https://stemangiola.github.io/bioc_2020_tidytranscriptomics/)  | [![GH](images/GitHub-Mark-32px.png)](https://github.com/stemangiola/bioc_2020_tidytranscriptomics) | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/stemangiola/bioc_2020_tidytranscriptomics) |
| [Public Data Resources in Bioconductor](https://waldronlab.io/PublicDataResources/)                                      | [![GH](images/GitHub-Mark-32px.png)](https://github.com/waldronlab/PublicDataResources)            | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/waldronlab/publicdataresources)            |
| [Annotating inter-sample DNA methylation and ATAC-seq variation with COCOA](http://databio.org/cocoa_workshop_bioc2020/) | [![GH](images/GitHub-Mark-32px.png)](https://github.com/databio/cocoa_workshop_bioc2020)           | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/databio/cocoa_workshop_bioc2020)           |
| [Human RNA-seq data from recount2 and related packages](https://github.com//LieberInstitute/recountWorkshop2020)         | [![GH](images/GitHub-Mark-32px.png)](https://github.com/LieberInstitute/recountWorkshop2020)       | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/lcollado/recountworkshop2020)              |
| [Cloud-based Genomics using Terra/AnVIL](https://waldronlab.github.io/AnVILWorkshop/)                                    | [![GH](images/GitHub-Mark-32px.png)](https://github.com/waldronlab/AnVILWorkshop)                  | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/shbrief/anvilworkshop)                     |
| [Epidemiology for Bioinformaticians Workshop](https://cmirzayi.github.io/EpiForBioWorkshop2020/)                         | [![GH](images/GitHub-Mark-32px.png)](https://github.com/cmirzayi/EpiForBioWorkshop2020)            | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/cmirzayi/epiforbioworkshop2020)            |

<sup></sup>

### 200 Level Workshops

| Workshop                                                                                                                                                                                                       | Source                                                                                                     | Docker                                                                                                                         |
| :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :--------------------------------------------------------------------------------------------------------- | :----------------------------------------------------------------------------------------------------------------------------- |
| [Best practices for ATAC-seq QC and data analysis](https://haibol2016.github.io/ATACseqQCWorkshop/)                                                                                                            | [![GH](images/GitHub-Mark-32px.png)](https://github.com/haibol2016/ATACseqQCWorkshop)                      | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/nemat1976/atacseqqcworkshop_2020)                 |
| [Interactive Visualization of SummarizedExperiment Objects with iSEE](https://iSEE.github.io/iSEEWorkshop2020)                                                                                                 | [![GH](images/GitHub-Mark-32px.png)](https://github.com/iSEE/iSEEWorkshop2020)                             | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/iseedevelopers/iseeworkshop2020)                  |
| [Functional enrichment analysis of high-throughput omics data](https://waldronlab.github.io/enrichOmics/)                                                                                                      | [![GH](images/GitHub-Mark-32px.png)](https://github.com/waldronlab/enrichOmics)                            | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/ludwigg/enrichomics)                              |
| [Copy number variation analysis with Bioconductor](https://waldronlab.github.io/CNVWorkshop)                                                                                                                   | [![GH](images/GitHub-Mark-32px.png)](https://github.com/waldronlab/CNVWorkshop)                            | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/ludwigg/cnvworkshop)                              |
| [MultiAssayExperiment and curatedTCGAData Bioconductor 2020 Workshop](https://waldronlab.github.io/MultiAssayWorkshop/)                                                                                        | [![GH](images/GitHub-Mark-32px.png)](https://github.com/waldronlab/MultiAssayWorkshop)                     | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/mr148/multiassayworkshop)                         |
| [CRISPRseek and GUIDEseq for designing target-specific gRNAs for the CRISPR genome editing system including base editor and prime editor](https://LihuaJulieZhu.github.io/CRISPRseekGUIDEseqBioc2020Workshop/) | [![GH](images/GitHub-Mark-32px.png)](https://github.com/LihuaJulieZhu/CRISPRseekGUIDEseqBioc2020Workshop/) | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/lihuajuliezhu/crisprseekguideseqbioc2020workshop) |
| [Integrated ChIP-seq analysis workshop](https://hukai916.github.io/IntegratedChIPseqWorkshop/)                                                                                                                 | [![GH](images/GitHub-Mark-32px.png)](https://github.com/hukai916/IntegratedChIPseqWorkshop)                | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/hukai916/integratedchipseqanalysis_workshop)      |
| [An introduction to matrix factorization and principal component analysis in R](https://aedin.github.io/PCAworkshop/)                                                                                          | [![GH](images/GitHub-Mark-32px.png)](https://github.com/aedin/PCAworkshop)                                 | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/aculhane/pcabiocworkshop2020)                     |
| [A workshop on discovering biomarkers from high through put response screens](https://bhklab.github.io/bioc2020workshop)                                                                                       | [![GH](images/GitHub-Mark-32px.png)](https://github.com/bhklab/bioc2020workshop)                           | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/bhklab/gxbioc2020workshop)                        |
| [Importing alevin scRNA-seq counts into R/Bioconductor](https://mikelove.github.io/alevin2bioc/)                                                                                                               | [![GH](images/GitHub-Mark-32px.png)](https://github.com/mikelove/alevin2bioc)                              | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/mikelove/alevin2bioc)                             |

<sup></sup>

### 500 Level Workshops

| Workshop                                                                                                                                               | Source                                                                                             | Docker                                                                                                                  |
| :----------------------------------------------------------------------------------------------------------------------------------------------------- | :------------------------------------------------------------------------------------------------- | :---------------------------------------------------------------------------------------------------------------------- |
| [Material on how to create and submit a package to Bioconductor](https://kayla-morrell.github.io/CreateAPackage/)                                      | [![GH](images/GitHub-Mark-32px.png)](https://github.com/Kayla-Morrell/CreateAPackage)              | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/kaylainterdonato01/createapackage)         |
| [Bioconductor toolchain for usage and development of reproducible bioinformatics pipelines in CWL](https://Liubuntu.github.io/Bioc2020RCWL)            | [![GH](images/GitHub-Mark-32px.png)](https://github.com/Liubuntu/Bioc2020RCWL)                     | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/liubuntu/bioc2020rcwl)                     |
| [Effectively Using the DelayedArray Framework to Support the Analysis of Large Datasets](https://petehaitch.github.io/BioC2020_DelayedArray_workshop/) | [![GH](images/GitHub-Mark-32px.png)](https://github.com/PeteHaitch/BioC2020_DelayedArray_workshop) | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/petehaitch/bioc2020_delayedarray_workshop) |
| [Detection of Differentially Interacting Chromatin Regions From Multiple Hi-C Datasets](https://mdozmorov.github.io/HiCcompareWorkshop/)               | [![GH](images/GitHub-Mark-32px.png)](https://github.com/mdozmorov/HiCcompareWorkshop)              | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/mdozmorov/hiccompareworkshop)              |
| [Trajectory inference across conditions: differential expression and differential progression](https://kstreet13.github.io/bioc2020trajectories)       | [![GH](images/GitHub-Mark-32px.png)](https://github.com/kstreet13/bioc2020trajectories)            | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/kstreet13/bioc2020trajectories)            |
| [Efficient, Scalable, and Reproducible Enrichment Workflows](https://montilab.github.io/hypeR-workshop/)                                               | [![GH](images/GitHub-Mark-32px.png)](https://github.com/montilab/hypeR-workshop)                   | [![Docker](images/docker_icon.png)](https://hub.docker.com/repository/docker/montilab/hyperworkshop)                    |

<sup></sup>

Workshops with preinstalled dependencies can be run locally using
Docker.

  - `docker pull <userid/workshop:latest>` will pull the latest image
    for a given `<workshop>`.
  - `docker run -e PASSWORD=<yourpassword> -p 8787:8787 -d --rm
    <userid/workshop>` will publish a container’s port 8787 to the host
    (`-p`), run in the detached mode (`-d`), and cleanly remove the
    container when it is stopped (`--rm`).
  - Open <http://localhost:8787> and login with username `rstudio` and
    password `<yourpassword>`.
  - Run `browseVignettes(package = "<workshop>")`. Click on one of the
    links, “HTML”, “source”, “R code”.
      - In case of “`The requested page was not found`” error, add
        `help/` to the URL right after the hostname. This is a [known
        bug](https://github.com/rocker-org/rocker-versioned/issues/178).

Advanced: Use `-v $(pwd):/home/rstudio` argument to map your local
directory to the container. Use `-e DISABLE_AUTH=true`, if you want
passwordless login to RStudio. On Windows, you may need to provide your
localhost IP address like `http://191.163.92.108:8787/` - find it using
`docker-machine ip default` in Docker’s terminal.

Example:

    docker pull waldronlab/publicdataresources:latest
    docker run -e PASSWORD=bioconductor -p 8787:8787 -d --rm waldronlab/publicdataresources
    # Open http://localhost:8787 and login using rstudio/bioconductor credentials
    # Run browseVignettes(package = "PublicDataResources")
    # Open http://localhost:8787/help/library/PublicDataResources/doc/PublicDataResources.html
    docker ps -a # List all running containers
    docker stop <CONTAINER ID> # or, <NAMES> - Stop a container

Workshop packages were created using the
[BuildABiocWorkshop2020](https://github.com/seandavi/BuildABiocWorkshop2020)
template.

*NOTE*. All workshops should include a valid `DESCRIPTION` file with the
following fields:

  - `URL:` - the GitHub pages URL
    (`https://username.github.io/repository`)
  - `DockerImage:` - the Docker repository slug in lowercase with
    (optional) tag (`username/repository:tag`)
